climada/engine/supplychain.py

```diff
@@ -0,0 +1,345 @@
+    Tables (MRIOT) and computing direct, indirect and total impacts.
+
+    Attributes
+    ----------
+    mriot_data : np.array
+        The input-output table data.
+    mriot_reg_names : np.array
+        Names of regions considered in the input-output table.
+    sectors : np.array
+        Sectors considered in the input-output table.
+    total_prod : np.array
+        Countries' total production.
+    mriot_type : str
+        Type of the adopted input-output table.
+    reg_pos : dict
+        Regions' positions within the input-output table and impact arrays.
+    reg_dir_imp : list
+        Regions undergoing direct impacts.
+    years : np.array
+        Years of the considered hazard events for which impact is calculated.
+    direct_impact : np.array
+        Direct impact array.
+    direct_aai_agg : np.array
+        Average annual direct impact array.
+    indirect_impact : np.array
+        Indirect impact array.
+    indirect_aai_agg : np.array
+        Average annual indirect impact array.
+    total_impact : np.array
+        Total impact array.
+    total_aai_agg : np.array
+        Average annual total impact array.
+    io_data : dict
+        Dictionary with the coefficients, inverse and risk_structure matrixes and
+        the selected input-output modeling approach.
+    """
+
+    def __init__(self):
+        """Initialize SupplyChain."""
+        self.mriot_data = np.array([], dtype='f')
+        self.mriot_reg_names = np.array([], dtype='str')
+        self.sectors = np.array([], dtype='str')
+        self.total_prod = np.array([], dtype='f')
+        self.mriot_type = ''
+        self.reg_pos = {}
+        self.years = np.array([], dtype='f')
+        self.direct_impact = np.array([], dtype='f')
+        self.direct_aai_agg = np.array([], dtype='f')
+        self.indirect_impact = np.array([], dtype='f')
+        self.indirect_aai_agg = np.array([], dtype='f')
+        self.total_impact = np.array([], dtype='f')
+        self.total_aai_agg = np.array([], dtype='f')
+        self.io_data = {}
+
+    def read_wiod16(self, year=2014, table_info_pos=None):
+        """Read multi-regional input-output tables of the 2016 release of the
+        WIOD project: http://www.wiod.org/database/wiots16
+
+        Parameters
+        ----------
+        year : int
+            Year of WIOD table to use. Valid years go from 2000 to 2014.
+            Default year is 2014.
+        table_info_pos: dict
+            Dictionary with info about how to properly read the WIOD table.
+            Default is None. Required keys are:
+                range_rows : tuple
+                    initial and end positions of data along rows.
+                range_cols : tuple
+                    initial and end positions of data along columns.
+                col_iso3 : int
+                    column with countries names in ISO3 codes.
+                col_sectors : int
+                    column with sector names.
```

Comment:
Strictly speaking, a dictionary with required keys as an argument is pretty bad style. 😉  🙊

Reply:
I am not to attached to any of the two options: I just didn't want too many kwargs; but can do either...

Reply:
I actually also tought that dictionaries are a good way to reduce the number of kwargs in certain cases :D

Reply:
I did actually not. 🤔 The parameters of a method should be laid out plain in the open and not hidden in a dictionary. Reducing argument number with a dictionary is basically cheating and makes the situation worse not better. No complexity is reduced, just moved out of sight. Leaving the developer using the method with the burden to figure out the ''real" signature. How? Docstring? Often not in sync. Code? Often hard to read.\
High number of arguments is a hint that something in the design is suboptimal. That's the first place to look into. 
In the case at hand, all of the `table_info` arguments _could_ be derived from the Excel file itself. That would be the highest potential when it comes to reducing argument numbers.\
However, deriving these figures takes some effort and there are situations where the combination of multiple arguments into a single one makes sense. But then, _please_ use a data class instead of a dictionary. Because a dictionary is _per se_ unrestricted and unknown and a data class is not.\
In the present situation the definition of a data class is probably over the top. There are "only" 5 arguments after all.

Reply:
Then I probably misunderstood you at some point. Thanks for the detailed reply!

Reply:
many thanks @emanuel-schmid - dictionary removed in commit https://github.com/CLIMADA-project/climada_python/pull/180/commits/eedc8ae0ece66cb2c38d5398b04849ec9da56c82.